src/Form/Button.cpp

```diff
@@ -150,6 +150,9 @@ Button::OnKeyCheck(unsigned key_code) const
 {
   switch (key_code) {
   case KEY_RETURN:
+  #ifdef ANDROID
+  case KEYCODE_BUTTON_L1:
```

Comment:
This change is not explained. What is `BUTTON_L1` and why do you want it to act as replacement for `RETURN`?

Reply:
Gamepads/Controllers by convention use the primary buttons (BUTTON_L1 & BUTTON_R1) to control 'click/select' and 'escape/back' UI behaviour. Modal dialogs in XCS currently assume these behaviours are linked to KEY_ENTER and KEY_ESCAPE; BUTTON_L1 & BUTTON_R1 are their controller analogues.

Is there a better place for this explanation?

Reply:
Yes, explanations belong in code comments (or commit messages). The code must be comprehensible without any network connection, only with the data in your local git repository.
After your explanation, I think it's a bad idea to hook `BUTTON_L1`/`R1` into each and every code location where we currently check `RETURN`/`ESCAPE`. **If** we decide that `BUTTON_L1` acts as a full-featured `RETURN` replacement *everywhere*, then we should translate this key code earlier. For example here: https://github.com/XCSoar/XCSoar/blob/master/src/Android/EventBridge.cpp#L47-L63

Reply:
I totally agree that is far better than my approach - I will rework the code to use this method.